packages/models/src/purpose/purposeReadModelAdapter.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  PurposeReadModel,
+  PurposeVersionDocumentReadModel,
+  PurposeVersionReadModel,
+} from "../read-models/purposeReadModel.js";
+import { Purpose, PurposeVersion, PurposeVersionDocument } from "./purpose.js";
+
+export const toReadModelPurposeVersionDocument = (
+  purposeVersionDocument: PurposeVersionDocument
+): PurposeVersionDocumentReadModel => ({
+  ...purposeVersionDocument,
+  createdAt: purposeVersionDocument.createdAt.toISOString(),
+});
+
+export const toReadModelPurposeVersion = (
+  purposeVersion: PurposeVersion
+): PurposeVersionReadModel => ({
+  ...purposeVersion,
+  createdAt: purposeVersion.createdAt.toISOString(),
+  updatedAt: purposeVersion.updatedAt?.toISOString(),
+  firstActivationAt: purposeVersion.firstActivationAt?.toISOString(),
+  suspendedAt: purposeVersion.suspendedAt?.toISOString(),
+  expectedApprovalDate: purposeVersion.expectedApprovalDate?.toISOString(),
+  riskAnalysis: purposeVersion.riskAnalysis
+    ? toReadModelPurposeVersionDocument(purposeVersion.riskAnalysis)
+    : undefined,
+});
+
+export const toReadModelPurpose = (purpose: Purpose): PurposeReadModel => ({
+  ...purpose,
+  createdAt: purpose.createdAt.toISOString(),
+  updatedAt: purpose.updatedAt?.toISOString(),
+  versions: purpose.versions.map(toReadModelPurposeVersion),
+});
```

Comment:
For the fields that can be `undefined`, why not doing the following?
```
updatedAt: purpose.updatedAt ? purpose.updatedAt.toISOString() : undefined
```

Reply:
They do the same thing, it's a matter of personal preference

Reply:
2 cents worth almost nothing: I prefer the shorter version `x?.y`, to me it is more readable (and being shorter there is a chance to avoid additional lines, which helps my OCD)